File: leetcode/100-999/169. Majority Element/solution_1.cpp
class Solution 
{
    public:
        int majorityElement(vector<int>& nums) 
        {
            int majority;
            map<int, int> numAndCount;

            for (int i = 0; i < size(nums); ++i)
            {
                numAndCount[nums[i]]++;

                if (numAndCount[nums[i]] > nums.size() / 2)
                {
                    majority = nums[i];
                    break;
                }
            }

            return majority;
        }
};

File: leetcode/100-999/169. Majority Element/solution_1.java
class Solution 
{
    public int majorityElement(int[] nums) 
    {
        int majority = -1;
        Map<Integer, Integer> numAndCount = new HashMap<>();

        for (int i = 0; i < nums.length; ++i)
        {
            numAndCount.put(nums[i], numAndCount.getOrDefault(nums[i], 0) + 1);

            if (numAndCount.get(nums[i]) > nums.length / 2)
            {
                majority = nums[i];
                break;
            }
        }

        return majority;
    }
}